apps/prairielearn/src/pages/administratorQuery/administratorQuery.ts

```diff
@@ -1,72 +1,104 @@
-const express = require('express');
+import express = require('express');
 const router = express.Router();
-const asyncHandler = require('express-async-handler');
-const fsPromises = require('fs').promises;
-const path = require('path');
-const _ = require('lodash');
-const hljs = require('highlight.js');
-const { stringify } = require('@prairielearn/csv');
+import asyncHandler = require('express-async-handler');
+import fsPromises = require('node:fs/promises');
+import path = require('path');
+import hljs from 'highlight.js';
+import { stringify } from '@prairielearn/csv';
+import { z } from 'zod';
 
-const jsonLoad = require('../../lib/json-load');
-const sqldb = require('@prairielearn/postgres');
+import jsonLoad = require('../../lib/json-load');
+import sqldb = require('@prairielearn/postgres');
+import {
+  AdministratorQuery,
+  AdministratorQueryRunParams,
+  AdministratorQueryResult,
+} from './administratorQuery.html';
 
 const sql = sqldb.loadSqlEquiv(__filename);
 
 const queriesDir = path.resolve(__dirname, '..', '..', 'admin_queries');
 const schemaFilename = path.resolve(__dirname, '..', '..', 'schemas', 'schemas', 'adminQuery.json');
 
+const AdministratorQueryQueryRunSchema = z.string();
+const AdministratorQueryResultSchema = z.object({
+  rows: z.array(z.object({})),
+  rowCount: z.number(),
+  columns: z.array(z.string()),
+});
+
 router.get(
   '/:query',
   asyncHandler(async (req, res, next) => {
-    res.locals.jsonFilename = req.params.query + '.json';
-    res.locals.sqlFilename = req.params.query + '.sql';
+    const jsonFilename = req.params.query + '.json';
+    const sqlFilename = req.params.query + '.sql';
 
-    res.locals.info = await jsonLoad.readJSONAsync(path.join(queriesDir, res.locals.jsonFilename));
+    const info = await jsonLoad.readJSONAsync(path.join(queriesDir, jsonFilename));
     const schema = await jsonLoad.readJSONAsync(schemaFilename);
-    await jsonLoad.validateJSONAsync(res.locals.info, schema);
-    res.locals.sql = await fsPromises.readFile(path.join(queriesDir, res.locals.sqlFilename), {
+    await jsonLoad.validateJSONAsync(info, schema);
```

Comment:
It'd be ideal to replace the use of a JSON schema file with a Zod schema.

Reply:
Resolved in comment 2f39462.